Marlin/src/gcode/calibrate/G34_M422.cpp

```diff
@@ -108,6 +108,9 @@ void GcodeSuite::G34() {
   }
 
   #if ENABLED(Z_STEPPER_AUTO_ALIGN)
+
+    probe.use_probing_tool();
```

Comment:
It is somewhat preferred to only change to the probing tool at the place below, after it's confirmed that the procedure will go forward, and then to restore the tool before re-enabling leveling. An extra restore of the original tool is fine at the end, since the tool-change function can check that the tool is already selected and exit doing nothing. The use of the nesting counter does ensure that only one restore will occur, and only after a save of the prior tool has been done, but perhaps doesn't need to increment when the tool is not changing.